app/services/payment_requests/payments/moneyhash_service.rb

```diff
@@ -0,0 +1,222 @@
+# frozen_string_literal: true
+
+module PaymentRequests
+  module Payments
+    class MoneyhashService < BaseService
+      include Customers::PaymentProviderFinder
+
+      PENDING_STATUSES = %w[PENDING].freeze
+      SUCCESS_STATUSES = %w[PROCESSED].freeze
+      FAILED_STATUSES = %w[FAILED].freeze
+
+      def initialize(payable = nil)
+        @payable = payable
+
+        super(nil)
+      end
+
+      def create
+        result.payable = payable
+        result.single_validation_failure!(error_code: "payment_method_error") if moneyhash_payment_method.nil?
+        return result unless should_process_payment?
+
+        unless payable.total_amount_cents.positive?
+          update_payable_payment_status(payment_status: :succeeded)
+          return result
+        end
+
+        payable.increment_payment_attempts!
+
+        moneyhash_result = create_moneyhash_payment
+        return result unless moneyhash_result
+
+        payment = Payment.new(
+          payable: payable,
+          payment_provider_id: moneyhash_payment_provider.id,
+          payment_provider_customer_id: customer.moneyhash_customer.id,
+          amount_cents: payable.amount_cents,
+          amount_currency: payable.currency&.upcase,
+          provider_payment_id: moneyhash_result.dig("data", "id"),
+          status: moneyhash_result.dig("data", "status")
+        )
+
+        payment.save!
+
+        payable_payment_status = payable_payment_status(payment.status)
+
+        update_payable_payment_status(
+          payment_status: payable_payment_status,
+          processing: payment.status == "processing"
+        )
+        update_invoices_payment_status(
+          payment_status: payable_payment_status,
+          processing: payment.status == "processing"
+        )
+        result.payment = payment
+        result.payable_payment_status
+        result
+      end
+
+      def update_payment_status(organization_id:, provider_payment_id:, status:, metadata: {})
+        payment_obj = Payment.find_or_initialize_by(provider_payment_id: provider_payment_id)
+        payment = if payment_obj.persisted?
+          payment_obj
+        else
+          create_payment(provider_payment_id:, metadata:)
+        end
+
+        return handle_missing_payment(organization_id, metadata) unless payment
+
+        result.payment = payment
+        result.payable = payment.payable
+        return result if payment.payable.payment_succeeded?
+        payment.update!(status:)
+
+        processing = status == "processing"
+        payment_status = payable_payment_status(status)
+        update_payable_payment_status(payment_status:, processing:)
+        update_invoices_payment_status(payment_status:, processing:)
+
+        PaymentRequestMailer.with(payment_request: payment.payable).requested.deliver_later if result.payable.payment_failed?
+        result
+      rescue BaseService::FailedResult => e
+        result.fail_with_error!(e)
```

Comment:
We recently fixed a bug where emails weren't sent if we couldn't create the payment, I think you want the same.
https://github.com/getlago/lago-api/pull/3155/files